ballerina-tests/http-service-tests/tests/enum_types_for_query_param_tests.bal

```diff
@@ -0,0 +1,147 @@
+
+public enum Status {
+    OLD,
+    NEW
+}
+
+public enum GENRE {
+    ROCK,
+    POP
+}
+
+public type NewStatus Status;
+public type UpdatedStatus NewStatus;
+
+service /enumQueryParam on generalListener {
+    resource function get album(Status status) returns string {
+        if status is OLD {
+            return "Old album";
+        }
+        return "New album";
+    }
+
+    resource function get artist(Status status, GENRE genre) returns string {
+            if status is OLD && genre is ROCK {
+                return "Old & Rock Artist";
+            }
+            return "Unknown Artist";
+        }
+
+    resource function get song(Status[] status) returns string {
+        if status[0] is OLD && status[1] is NEW {
+            return "Old & New Song";
+        }
+        return "Unknown Song";
+    }
+
+    resource function get newSong(NewStatus newStatus) returns string {
+        if newStatus is OLD {
+            return "Old Song";
+        }
+        return "Unknown Song";
+    }
+
+    resource function get updatedSong(UpdatedStatus updatedStatus) returns string {
+        if updatedStatus is OLD {
+            return "Old Song";
+        }
+        return "Unknown Song";
+    }
+}
+
+@test:Config {}
+function testEnumTypeForQueryParam() returns error? {
+    http:Client 'client = check new("http://localhost:9000");
+    string response = check 'client->get("/enumQueryParam/album?status=OLD");
+    test:assertEquals(response, "Old album");
+}
+
+@test:Config {}
+function testMulitpleEnumsForQueryParam() returns error? {
+    http:Client 'client = check new("http://localhost:9000");
+    string response = check 'client->get("/enumQueryParam/artist?status=OLD&genre=ROCK");
+    test:assertEquals(response, "Old & Rock Artist");
+}
+
+@test:Config {}
+function testEnumArrayTypeForQueryParam() returns error? {
+    http:Client 'client = check new("http://localhost:9000");
+    string response = check 'client->get("/enumQueryParam/song?status=OLD&status=NEW");
+    test:assertEquals(response, "Old & New Song");
+}
+
+@test:Config {}
+function testReferredEnumTypeForQueryParam() returns error? {
+    http:Client 'client = check new("http://localhost:9000");
+    string response = check 'client->get("/enumQueryParam/newSong?newStatus=OLD");
+    test:assertEquals(response, "Old Song");
+}
+
+@test:Config {}
+function testNestedReferredEnumTypeForQueryParam() returns error? {
+    http:Client 'client = check new("http://localhost:9000");
+    string response = check 'client->get("/enumQueryParam/updatedSong?updatedStatus=OLD");
+    test:assertEquals(response, "Old Song");
+}
+
+@test:Config {}
+function testNestedReferredInvalidEnumTypeForQueryParam() returns error? {
+    http:Client 'client = check new("http://localhost:9000");
+    string|error response = 'client->get("/enumQueryParam/updatedSong?updatedStatus=Unknown");
+    if response is error {
+        test:assertEquals(response.message(), "Bad Request");
```

Comment:
Rather than asserting the error message, can we get the response and assert the status code?
Check other places as well

Reply:
Changed in https://github.com/ballerina-platform/module-ballerina-http/pull/1424/commits/0921e4ac54c51c842fe9bbf9adf223e522c63699